amqp/src/main/scala/akka/stream/contrib/amqp/model.scala

```diff
@@ -18,46 +19,129 @@ sealed trait AmqpSourceSettings extends AmqpConnectorSettings
 final case class NamedQueueSourceSettings(
   connectionSettings: AmqpConnectionSettings,
   queue:              String,
-  declarations:       Seq[Declaration],
+  declarations:       Seq[Declaration]       = Seq.empty,
   noLocal:            Boolean                = false,
   exclusive:          Boolean                = false,
   consumerTag:        String                 = "default",
   arguments:          Map[String, AnyRef]    = Map.empty
-) extends AmqpSourceSettings
+) extends AmqpSourceSettings {
+  @annotation.varargs
+  def withDeclarations(declarations: Declaration*) = copy(declarations = declarations.toList)
+
+  def withNoLocal(noLocal: Boolean) = copy(noLocal = noLocal)
+
+  def withExclusive(exclusive: Boolean) = copy(exclusive = exclusive)
+
+  def withConsumerTag(consumerTag: String) = copy(consumerTag = consumerTag)
+
+  def withArguments(argument: (String, AnyRef), arguments: (String, AnyRef)*) =
+    copy(arguments = (argument +: arguments).toMap)
+
+  @annotation.varargs
+  def withArguments(argument: akka.japi.Pair[String, AnyRef], arguments: akka.japi.Pair[String, AnyRef]*) =
+    copy(arguments = (argument +: arguments).map(_.toScala).toMap)
+}
+
+object NamedQueueSourceSettings {
+  /**
+   * Java API
+   */
+  def create(connectionSettings: AmqpConnectionSettings, queue: String) =
+    NamedQueueSourceSettings(connectionSettings, queue)
+}
```

Comment:
Sorry that you had to do this double work, I have no idea where my pretty much identical work went.